Add Option type (#1232)
stellar/stellar-docs: docs/learn/encyclopedia/contract-development/types/built-in-types.mdx

:::tip

Custom types like structs, enums, and unions are also supported. See [Custom Types]. 

:::

> [Custom Types]: custom-types.mdx

Reviewer: I believe your build is failing because you need a blank line above this. It's a pain, I know. The `:::tip` kind of admonitions need some specific blank line padding:

- a blank line on either side of the opening `:::tip` tag
- a blank line on either side of the closing `:::` tag

```markdown
Some paragraph could be here. Then, I'll put a blank line:

:::tip

Blank lines around this part, too!

:::

This paragraph might talk about how there are so many blank lines to love!!
```

Reviewer: Oh, this isn't really related to your change, huh? Don't know why the markdown formatting thing decided to do it now... 🤷🏻‍♂️

Address can be used as a contract function input argument (for example, to identify the payment recipient), as a data key (for example, to store the balance), as the authentication & authorization source (for example, to authorize a token transfer) etc.

See [authorization documentation](../../security/authorization.mdx) for more details on how to use the `Address` type.

## Option (`Option`)

Option represents an optional value.

The Option type is used for values that may, or may not, be present. The Option type is an enum and can either be `Some` (some value exists) or `None` (no value exits).

> While Option acts like Rust's Option<T>, the Option type is not explicitly represented in XDR. The Option type is represented in the XDR by the `ScVal` `ScVoid` value when no value exists (None), any by any other `ScVal` value when the value exists (Some).

Reviewer: ```suggestion
While Option acts like Rust's `Option<T>`, the Option type is not explicitly represented in XDR. The Option type is represented in the XDR by the `ScVal` `ScVoid` value when no value exists (None), any by any other `ScVal` value when the value exists (Some).
```

I think this is probably the main culprit. Docusaurus is expecting `<T>` to be a react component or something. Then, when it can't find a closing `</T>` it freaks out

> While Option acts like Rust's Option<T>, the Option type is not explicitly represented in XDR. The Option type is represented in the XDR by the `ScSpecTypeOption` type when a value exits (Some), and by the type `ScVoid` when a value doesn't exist (None). This can be verified by inspecting the WASM with the `soroban contract inspect` command. 

Reviewer 2: This paragraph mixes up the spec XDR with the protocol XDR. The spec XDR is used only to describe the interface of a contract. Any of the types prefixed with ScSpec relate to the spec XDR. The ScVoid and ScVal type relates to the actual protocol for contracts.

```suggestion
While Option acts like Rust's Option<T>, the Option type is not explicitly represented in XDR. The Option type is represented in the XDR by the `ScVal` `ScVoid` value when no value exists (None), any by any other `ScVal` value when the value exists (Some).
```
